builtin/game/chat.lua

```diff
@@ -1,27 +1,40 @@
 -- Minetest: builtin/game/chat.lua
 
+-- Helper function made global for use in mods
```

Comment:
this needs to be documented if kept global
but I don't think it makes much sense to expose anyway

Reply:
> this needs to be documented if kept global

> Note that I'm yet to add documentation for core.safe_gsub. This PR is otherwise ready for review.

> but I don't think it makes much sense to expose anyway

Wouldn't mods benefit from this function when faced with a similar to use this for similar purposes? We do expose many other useful functions for mods this way, like `table.indexof` and the like.

Reply:
Our `safe_gsub` only does one substitution at a time. Modders may want behaviour that is closer to the original instead.

Reply:
You're right, I'll make the function local.